source/connection.c

```diff
@@ -478,6 +474,7 @@ struct aws_http_server *aws_http_server_new(const struct aws_http_server_options
         goto hash_table_error;
     }
 
+    s_server_lock_synced_data(server);
```

Comment:
comment `/* Protect against callbacks firing before server->socket is set */`

Reply:
I have sad sad news.
While I wrote this, I wondered if this actually did protect us, and it doesn't. The accept callback accesses server->socket->... multiple times for logging before it ever acquires a lock. Let's do the following:
1) Log about a server based on `(void*)server` instead of `server->socket->address`.
2) Put another comment somewhere in the accept callback like `/* Be careful not to access server->socket until lock is acquired to avoid race conditions */`